src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -8026,6 +8026,793 @@ .maxstack  4
+        [InlineData(LanguageVersion.CSharp10)]
+        [InlineData(LanguageVersion.CSharp11)]
+        public void Deconstruct_02A(LanguageVersion languageVersion)
+        {
+            var source =
+@"ref struct R1
+{
+    public R1(ref int i) { }
+    public void Deconstruct(out int x, out R2 y) => throw null;
+}
+ref struct R2
+{
+    public R2(ref int i) { }
+}
+class Program
+{
+    static void F1()
+    {
+        int i = 1;
+        var r = new R1(ref i);
+        int x1;
+        R2 y1;
+        (x1, y1) = r; // 1
+        r.Deconstruct(out x1, out y1); // 2
+    }
+    static void F2()
+    {
+        int i = 2;
+        var r = new R1(ref i);
+        int x2;
+        R2 y2 = new R2(ref i);
+        (x2, y2) = r;
+        r.Deconstruct(out x2, out y2);
+    }
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular.WithLanguageVersion(languageVersion));
+            if (languageVersion == LanguageVersion.CSharp10)
+            {
+                comp.VerifyEmitDiagnostics();
+            }
+            else
+            {
+                comp.VerifyEmitDiagnostics(
+                    // (18,9): error CS8352: Cannot use variable '(x1, y1) = r' in this context because it may expose referenced variables outside of their declaration scope
+                    //         (x1, y1) = r; // 1
+                    Diagnostic(ErrorCode.ERR_EscapeVariable, "(x1, y1) = r").WithArguments("(x1, y1) = r").WithLocation(18, 9),
+                    // (18,20): error CS8350: This combination of arguments to 'R1.Deconstruct(out int, out R2)' is disallowed because it may expose variables referenced by parameter 'this' outside of their declaration scope
+                    //         (x1, y1) = r; // 1
+                    Diagnostic(ErrorCode.ERR_CallArgMixing, "r").WithArguments("R1.Deconstruct(out int, out R2)", "this").WithLocation(18, 20),
+                    // (19,9): error CS8352: Cannot use variable 'r' in this context because it may expose referenced variables outside of their declaration scope
+                    //         r.Deconstruct(out x1, out y1); // 2
+                    Diagnostic(ErrorCode.ERR_EscapeVariable, "r").WithArguments("r").WithLocation(19, 9),
+                    // (19,9): error CS8350: This combination of arguments to 'R1.Deconstruct(out int, out R2)' is disallowed because it may expose variables referenced by parameter 'this' outside of their declaration scope
+                    //         r.Deconstruct(out x1, out y1); // 2
+                    Diagnostic(ErrorCode.ERR_CallArgMixing, "r.Deconstruct(out x1, out y1)").WithArguments("R1.Deconstruct(out int, out R2)", "this").WithLocation(19, 9));
+            }
+        }
+
+        [WorkItem(64448, "https://github.com/dotnet/roslyn/issues/64448")]
+        [Theory]
+        [InlineData(LanguageVersion.CSharp10)]
+        [InlineData(LanguageVersion.CSharp11)]
+        public void Deconstruct_02B(LanguageVersion languageVersion)
```

Comment:
Same feedback here.

Reply:
Removed 2A and 2B (the equivalent Span<T> cases were 4A and 4B).